generators/csharp/codegen/src/project/CsharpProject.ts

```diff
@@ -71,6 +71,16 @@ export class CsharpProject {
             this.coreFiles.push(await this.createCoreAsIsFile(file));
         }
 
+        const githubWorkflowTemplate = (await readFile(getAsIsFilepath(AsIsFiles.CiYaml))).toString();
+        const githubWorkflow = template(githubWorkflowTemplate)({
+            projectName: this.name,
+            shouldWritePublishBlock: this.context.publishConfig != null,
+            nugetTokenEnvvar: this.context.publishConfig?.apiKeyEnvironmentVariable ?? "NUGET_API_TOKEN"
+        }).replaceAll("\\{", "{");
+        const ghDir = join(this.absolutePathToOutputDirectory, RelativeFilePath.of(".github/workflows"));
+        await mkdir(ghDir, { recursive: true });
+        await writeFile(join(ghDir, RelativeFilePath.of("ci.yml")), githubWorkflow);
```

Comment:
how do we make sure this only happens when the generator is in `github` mode?

Reply:
The template has a conditional that's if the publish block is present, which is a GitHub output block

I tested it and it works well, the whole block is added or removed

Check the template evaluation a couple lines above

Reply:
https://github.com/fern-api/fern/pull/3621/files#diff-2e2d2dd0dc8bd56571b1e52014eab736a44e2dc8f6d0f4306c98f7ababb197daR44-R53